src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSchedulesController.java

```diff
@@ -163,6 +163,14 @@ public PersonalSchedule putScheduleById(
     personalschedule.setDescription(incomingSchedule.getDescription());
     personalschedule.setQuarter(incomingSchedule.getQuarter());
 
+    Optional<PersonalSchedule> existCheck =
```

Comment:
You may need to use a different look up when editing vs when creating a new one.

When editing, you need a search that will return multiple elements (e.g. a List<> or Iterable<> rather than an Optional<>, because the search may include the existing record.

Or, you could write a custom search using SQL that takes a name and the id of the existing record, and looks for something like this (the syntax may be wrong, but this is the general idea):

```sql
select *
from PersonalSchedules
where name=$name and id<>$id
```